server/chat-plugins/helptickets.ts

```diff
@@ -636,6 +697,134 @@ const cheatingScenarios = [
 	],
 ];
 
+export const textTickets: {[k: string]: TextTicketInfo} = {
+	inapname: {
+		title: "What's the inappropriate username?",
+		checker(input) {
+			if (!Users.get(input)) {
+				return [
+					"Please specify a valid username - that name was not found.",
+					"Maybe you spelled it wrong?",
+				];
+			}
+			return true;
+		},
+		getReviewDisplay(ticket) {
+			let buf = ``;
+			if (!ticket.open) return buf;
+			if (ticket.meta?.startsWith('user-')) {
+				const tar = ticket.meta.slice(5);
+				buf += `<strong>Username: ${tar}</strong><br />`;
+				buf += `<form data-submitsend="/forcerename ${tar},{reason}">`;
+			} else {
+				buf += `<strong>Provide a username to forcerename:</strong><br />`;
+				buf += `<form data-submitsend="/msgroom staff,/forcerename {text},{reason}">`;
+				buf += `Name: <input name="text" /><br />`;
+			}
+			buf += `Reason (optional:) <input name="reason" /><br />`;
+			buf += `<br /><button class="button notifying" type="submit">Forcerename</button></form>`;
+			return buf;
+		},
+	},
+	battleharassment: {
+		title: "Please provide a link to the battle (taken from the \"Upload and share\" button or copied from the browser URL)",
+		checker(input) {
+			if (BATTLES_REGEX.test(input) || REPLAY_REGEX.test(input)) return true;
+			return ['Please provide at least one valid battle or replay URL.'];
+		},
+		getReviewDisplay(ticket, staff, connection) {
+			let buf = ``;
+			const [text, context] = ticket.text;
+			const rooms = getBattleLinks(text);
+
+			if (context) {
+				rooms.push(...getBattleLinks(context));
+			}
+
+			let battlelogNoticeAdded = false;
+			for (const [i, url] of rooms.entries()) {
+				if (rooms.indexOf(url) !== i) {
+					rooms.splice(i, 1);
+					continue;
+				}
+				const room = Rooms.get(url);
+				if (room) {
+					staff.joinRoom(room, connection);
```

Comment:
I'm not a fan of automatically joining battles when I take a ticket, especially for things like champ battles.

Reply:
Was explicitly requested. I can talk to Teclis about it.

Reply:
Yeah, me and Annika are concerned about it, I would assume other staff would be as well mostly due to champ battles.

Reply:
See #private-dev. We also really don't have that many champ reports anymore, I personally don't see it as much of an issue.

Reply:
Felucia also mentions its possibly exploitable by reporting battles with a ton of turns to freeze the browser.

Reply:
Fair enough.

Reply:
Done.